File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
         int windowSum=0;
        int minLen=Integer.MAX_VALUE;
        int windowStart=0;
        for (int windowEnd = 0; windowEnd < nums.length; windowEnd++) {
            windowSum+=nums[windowEnd];
            while (windowSum>=target){
                minLen=Math.min(minLen,windowEnd-windowStart+1);
                windowSum-=nums[windowStart];
                windowStart++;
            }
        }
        return (minLen==Integer.MAX_VALUE)?0:minLen;
    }
}

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        
        res = float('inf')
        left = 0
        curSum = 0
        for i in range(len(nums)):
            curSum += nums[i]
            while curSum >= target:
                res = min(res, i - left + 1)
                curSum -= nums[left]
                left += 1
        return res if res != float('inf') else 0
        